src/gtc/passes/gtir_legacy_extents.py

```diff
@@ -0,0 +1,63 @@
+from dataclasses import dataclass, field
+from typing import Any, Dict, Union
+
+from eve import NodeVisitor
+from eve.utils import XIterator
+from gt4py.definitions import Extent
+from gtc import gtir
+
+
+def _iter_field_names(node: Union[gtir.Stencil, gtir.ParAssignStmt]) -> XIterator[gtir.FieldAccess]:
+    return node.iter_tree().if_isinstance(gtir.FieldDecl).getattr("name").unique()
+
+
+def _iter_assigns(node: gtir.Stencil) -> XIterator[gtir.ParAssignStmt]:
+    return node.iter_tree().if_isinstance(gtir.ParAssignStmt)
+
+
+def _ext_from_off(offset: gtir.CartesianOffset) -> Extent:
+    return Extent(
+        ((min(offset.i, 0), max(offset.i, 0)), (min(offset.j, 0), max(offset.j, 0)), (0, 0))
+    )
+
+
+FIELD_EXT_T = Dict[str, Extent]
+
+
+class LegacyExtentsVisitor(NodeVisitor):
+    @dataclass
+    class AssignContext:
+        left_extent: Extent
+        assign_extents: FIELD_EXT_T = field(default_factory=dict)
+
+    def visit_Stencil(self, node: gtir.Stencil, **kwargs: Any) -> FIELD_EXT_T:
+        field_extents = {name: Extent.zeros() for name in _iter_field_names(node)}
+        for assign in _iter_assigns(node).to_list()[::-1]:
```

Comment:
I would prefer reverse iterating, it would eliminate the need to store a list.

Reply:
Nope, we can iterate depth first top down, depth first bottom up and breadth first top down but not right to left. XIterator is not reversible either. We can do `reversed(_iter_assigns(node).to_list())` which is more readable but otherwise equivalent.